### DISCUSSION TITLE
Doesn't work on catalina through php exec()

### DISCUSSION THREAD

**Discussion by hebertb** [ImageMagick/ImageMagick#1945] [General]
Catalina 10.15 PHP 7.3.8 ImageMajick works in terminal when I run the command line below, but does not work in the php exec(), even though I set the environment variables ($MAGICK_HOME, $DYLD_LIBRARY_PATH) through putenv() and apache_setenv().

/usr/local/bin/convert "/Library/FileMaker Server/HTTPServer/htdocs/Proof/jpg/2020/wk17/23227_Tabloid/CONUS/20wk17_L.pdf" -colorspace rgb -units PixelsPerInch -resize 122x -density 72 -background white -alpha remove -scene 1 "/Library/FileMaker Server/HTTPServer/htdocs/Proof/jpg/2020/wk17/23227_Tabloid/CONUS/20wk17_L_pg%02d.jpg" 2>&1

convert: no images defined `/Library/FileMaker Server/HTTPServer/htdocs/Proof/jpg/2020/wk17/23227_Tabloid/CONUS/20wk17_L_pg%02d.jpg' @ error/convert.c/ConvertImageCommand/3282

**Comment by snibgo** [comment]
Test a simpler command that does not involve PDF files, such as "magick rose: r.png". Does that work?

Does IM work to convert the PDF from the command line?

If those work, the problem may be that your PHP doesn't permit IM to call Ghostcript to rasterize the PDF. Perhaps this is cured by modifying delegates.xml to use the full path to GS.

**Comment by fmw42** [comment]
As snob said, you may have to edit your ImageMagick delegates.xml file to put the full path the `gs` in the PS/PDF/etc commands. Edit all commands that decode using `gs` to include the full path to Ghostscript where you find `gs`

`  <delegate decode="pdf" encode="eps" mode="bi" command="&quot;gs&quot; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 -sPDFPassword=&quot;%a&quot; &quot;-sDEVICE=eps2write&quot; &quot;-sOutputFile=%o&quot; &quot;-f%i&quot;"/>
`

**Reply by hebertb** [reply]
Did that...no joy.  Seems to be looking  for the dyld library, even though I set that path using putenv(); It still looks for it in /ImageMagick-7.0.10/lib/libMagickCore-7.Q16HDRI.7.dylib

**Reply by fmw42** [reply]
Where was the delegates.xml file edited? Was it in your download directory or in etc/ImageMagick-7. Be sure to do it in /etc/....

**Reply by hebertb** [reply]
nope /etc. I give up. I've been using it for 5 years without issue on any OS until Catalina. Gonna try backing down to Mojave.

**Comment by hebertb** [comment]
Well I discovered that it didn't install all of the libraries by using otool -L /path/to/convert. I used the source code to do the install since I was unable to use homebrew to install it due to the tight security restrictions I had to deal with. I copied all the of libraries to the expected directory and it started working as expected. I used the ./configure, make, then make install to install it, but apparently it didn't install like it should've. Hope this helps someone down the road.

**Reply by hebertb** [reply]
Well I have the same problem with the next catalina machine. If I  run "convert ~/Desktop/test.pdf ~/Desktop/test%02d.jpg" in terminal, it doesn't work. If I run "export DYLD_LIBRARY_PATH=/path/to/lib" and then run "convert ~/Desktop/test.pdf ~/Desktop/test%02d.jpg" in terminal, it works fine. If I try and run it from php, using the setenv() to set the DYLD_LIBRARY_PATH, it errors out. Apparently setenv() on php 7.3.11 doesn't work. What can I do so that I don't have to export the DYLD path to get it to work?
